### PR TITLE
feat: update to webpack 5

### PR DESCRIPTION
**Pull request by scefali**
This PR upgrades Webpack to 5 and pins the Node version with Volta to `14.19.2`

### CODE REVIEW

File: config/webpack.demo.config.js [hunk 1]
```diff
@@ -38,15 +38,19 @@ module.exports = {
   },
   resolve: {
     extensions: ['.tsx', '.ts', '.js'],
+    fallback: {
+      path: false,
+      crypto: false,
```
Comment (scefali) [inline config/webpack.demo.config.js:43]: @evanpurkhiser Pretty sure we shouldn't fallback on these libraries cause I can't imagine why we'd need them on the browser lol

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@sentry-internal/global-search",
   "description": "JavaScript library and helper utilities for searching Sentry sites via Algolia.",
-  "version": "0.1.3",
+  "version": "0.2.0",
```
Comment (scefali) [inline package.json:4]: I figured we should at least increment the minor version with upgrading Webpack, maybe this should be 1.0.0 though

File: config/webpack.demo.config.js (hunk 1 above)
Comment (evanpurkhiser) [reply config/webpack.demo.config.js:43]: you'd be surprised, but this seems fin yeah

File: package.json [hunk 3]
```diff
@@ -86,5 +86,8 @@
       "node_modules",
       "sample-document.js"
     ]
+  },
+  "volta": {
+    "node": "14.19.2"
```
Comment (evanpurkhiser) [inline package.json:91]: We should be using the volta action in github actions then I think

**Pull request merged**